pulp_smash/cli.py

```diff
@@ -816,3 +819,149 @@ def _yum_apply_erratum(self, erratum):
     def apply_erratum(self, erratum):
         """Dispatch to proper _{self.name}_apply_erratum."""
         return getattr(self, '_{0}_apply_erratum'.format(self.name))(erratum)
+
+
+class RegistryClient:
+    """A container registry client on test runner machine.
+
+    Each instance of this class represents the registry client on a host. An
+    example may help to clarify this idea:
+
+    >>> from pulp_smash import cli, config
+    >>> registry = cli.RegistryClient(config.get_config())
+    >>> image = registry.pull('image_name')
+
+    In the example above, the ``registry`` object represents the client
+    on the host where pulp-smash is running the test cases.
+
+    Upon instantiation, a :class:`RegistryClient` object talks to its target
+    host and uses simple heuristics to determine which registry client is used.
+
+    Upon instantiation, this object determines whether it is running as root.
+    If not root, all commands are prefixed with "sudo".
+    Please ensure that Pulp Smash can either execute commands as root
+    or can successfully execute ``sudo`` on the local host.
```

Comment:
```suggestion
    or can successfully execute "sudo" on the localhost.
```

Reply:
I followed the same pattern we have in https://github.com/PulpQE/pulp-smash/blob/master/pulp_smash/cli.py#L705